File: awesome_places_api/AwesomePlaces.Core/Entities/Place.cs
using AwesomePlaces.Core.Enums;
using AwesomePlaces.Core.Exceptions;

namespace AwesomePlaces.Core.Entities;

public class Place
{
    public Guid Id { get; protected set; } = Guid.NewGuid();
    public string Name { get; init; }
    public string Description { get; set; }
    public PlaceTypes PlaceType { get; init; }
    public byte[] Image { get; init; }
    public double Rating { get; init; }
    public bool RequiredFee { get; init; }
    public virtual Coordinate Coordinate { get; init; }
    public virtual Address Address { get; init; }

    public Place() { }

    public Place(Guid id, string name, string description, PlaceTypes placeType, Coordinate coordinate, Address address, byte[] image, double rating, bool requiredFee)
    {
        Id = id;
        Name = name;
        Description = description;
        PlaceType = placeType;
        Coordinate = coordinate;
        Address = address;
        Image = image;
        Rating = rating;
        RequiredFee = requiredFee;
    }

    public void ChangeDescription(string description)
    {
        if (string.IsNullOrWhiteSpace(description))
        {
            throw new InvalidPlaceDescriptionException(description);
        }

        this.Description = description;
    }
}

File: src/AwesomePlaces.MobileUI/lib/src/features/explore/data/models/place.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:awesome_places/src/features/explore/data/models/place_types.dart';

import 'address.dart';
import 'coordinate.dart';

class Place {
  String id;
  String name;
  String description;
  PlaceTypes placeType;
  Uint8List image;
  num rating;
  bool requiredFee;
  Coordinate? coordinate;
  Address? address;

  Place({
    required this.id,
    required this.name,
    required this.description,
    required this.placeType,
    required this.image,
    required this.rating,
    required this.requiredFee,
    required this.address,
    required this.coordinate,
  });

  factory Place.fromJson(Map<String, dynamic> json) {
    return Place(
        id: json['id'] as String,
        name: json['name'] as String,
        description: json['description'] as String,
        placeType: PlaceTypes.values.elementAt(json['placeType']),
        image: base64.decode(json['image']),
        rating: json['rating'] as num,
        requiredFee: json['requiredFee'] as bool,
        address: Address.fromJson(json['address']),
        coordinate: Coordinate.fromJson(json['coordinate']));
  }

  static List<Place>? fromListJson(List<dynamic> json) {
    List<Place> places = <Place>[];

    for (var element in json) {
      Place place = Place.fromJson(element);
      places.add(place);
    }
    return places;
  }
}
